lib/Runtime/PlatformAgnostic/Platform/Unix/AssemblyCommon.cpp

```diff
@@ -0,0 +1,32 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "Common.h"
+#include "PlatformAgnostic/AssemblyCommon.h"
+
+extern void mac_fde_wrapper(const char *dataStart, mac_fde_reg_op reg_op)
+{
+    for(const char *head = dataStart, *last = dataStart + CC_FDE_SIZE;
+        head != last;
```

Comment:
I suggest avoid the `last` check. FDE is not of fixed size. The check may mislead some to think it is. The CIE/FDE data is guaranteed to terminate from the `break` in the loop body below. If for any reason the data is wrong and does not, it makes no difference whether it hangs here or crashes later on. Hanging here is better in my opinion because it is closer to the source of problem.